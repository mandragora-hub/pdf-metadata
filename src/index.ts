// import readingTime from "reading-time";
// import { getRawTextContent, getMetadataAndPages } from "./lib.js";

export default async function extractPDFMetadata(arrayBuffer: ArrayBuffer) {
  // const pdfBuffer = Buffer.from(arrayBuffer);
  // const text = await getRawTextContent(pdfBuffer);
  // const { pages, metadata } = await getMetadataAndPages(pdfBuffer);
  // const stats = readingTime(text);

  // return { pages: pages, wordCount: stats.words, readTime: stats.time, metadata };

  return "asdfsadf"
}
