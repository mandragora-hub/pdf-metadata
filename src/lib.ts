// import { PdfReader } from "pdfreader";
// import { PDFDocument } from "pdf-lib";

// export function getRawTextContent(pdfBuffer) {
//   const promise = new Promise((resolve, reject) => {
//     const pdfReader = new PdfReader();
//     const textArray: string[] = [];
//     pdfReader.parseBuffer(pdfBuffer, (err, item) => {
//       if (err) reject("error with pdfReader:", err);
//       else if (!item) {
//         const text = textArray.join(" ");
//         return resolve(text);
//       } else if (item.text) {
//         textArray.push(item.text);
//       }
//     });
//   });
//   return promise;
// }

// export function getMetadataAndPages(pdfBuffer) {
//   const promise = new Promise((resolve, reject) => {
//     PDFDocument.load(pdfBuffer, { updateMetadata: false, ignoreEncryption: true })
//       .then((pdfDoc) => {
//         const pages = pdfDoc.getPages();
//         resolve({
//           pages: pages.length,
//           metadata: {
//             title: pdfDoc.getTitle(),
//             author: pdfDoc.getAuthor(),
//             subject: pdfDoc.getSubject(),
//             creator: pdfDoc.getCreator(),
//             keywords: pdfDoc.getKeywords(),
//             producer: pdfDoc.getProducer(),
//             creationDate: pdfDoc.getCreationDate(),
//             modificationDate: pdfDoc.getModificationDate(),
//           },
//         });
//       })
//       .catch((err) => reject(err));
//   });
//   return promise;
// }
