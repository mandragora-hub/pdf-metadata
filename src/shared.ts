import fs from "fs";
export function isArrayBuffer(v: any) {
  return typeof v === "object" && v?.byteLength !== undefined;
}

export function isValidFetchUrl(url: string, baseUrl?: string) {
  try {
    const { protocol } = baseUrl != null ? new URL(url, baseUrl) : new URL(url);
    // The Fetch API only supports the http/https protocols, and not file/ftp.
    return protocol === "http:" || protocol === "https:";
  } catch {
    return false; // `new URL()` will throw on incorrect data.
  }
}

export function isValidUrl(urlString: string) {
  try {
    return Boolean(new URL(urlString));
  } catch (e) {
    return false;
  }
}

export function fileExists(path: string) {
  return fs.existsSync(path);
}
