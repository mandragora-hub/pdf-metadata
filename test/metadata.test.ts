import { describe, expect, test } from "@jest/globals";
import { loadDocument } from "../src/index";

const MAX_TIMEOUT = 40000;

describe("Get info of module", () => {
  test(
    "Load and get info document",
    async () => {
      const url = "https://litterarum.onrender.com/api/v1/files/el-hombre-que-calculaba-malba-tahan.pdf";
      const doc = await loadDocument(url);
      const info = await doc.getInfo();
      expect(info).toMatchSnapshot();
    },
    MAX_TIMEOUT,
  );
});
