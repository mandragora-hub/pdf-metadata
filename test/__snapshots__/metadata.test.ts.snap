// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get info of module Load and get info document 1`] = `
{
  "metadata": {
    "author": undefined,
    "creationDate": 2017-11-25T15:43:31.000Z,
    "creator": undefined,
    "keywords": undefined,
    "modificationDate": undefined,
    "producer": "PDF-XChange Printer 2012 ABBYY (5.5 build 308.2) [Windows 8 x64 (Build 9200)]",
    "subject": undefined,
    "title": undefined,
  },
  "pages": 225,
  "readTime": 20326200,
  "wordCount": 67754,
}
`;
